docs/proposals/multiple-dns-provider-support.md

```diff
@@ -0,0 +1,86 @@
+# Multiple DNS Provider Support
+
+Authors: Michael Nairn @mikenairn
+
+Epic: https://github.com/Kuadrant/multicluster-gateway-controller/issues/189
+
+Date: 25th May 2023
+
+
+## Job Stories
+
+- As a developer, I want to use MGC with a domain hosted in one of the major cloud DNS providers (Google Cloud DNS, Azure DNS or AWS Route53) 
+- As a developer, I want to use  multiple domains with a single instance of MGC, each hosted on different cloud providers
+
+## Goals
+
+- Add ManagedZone and DNSRecord support for [Google Cloud DNS](https://cloud.google.com/dns/docs/)
+- Add ManagedZone and DNSRecord support for [Azure DNS](https://azure.microsoft.com/en-us/services/dns)
+- Add DNSRecord support for [CoreDNS](https://coredns.io/) (Default for development environment)
```

Comment:
What's the reason for choosing this as the default for development environment?

Reply:
Also, why not ManagedZone support here?

Reply:
Using CoreDNS as an option is really an investigation task, will update.

Currently in order to actually do anything useful with MGC you have to have a Route53 account and your own domain registered with it, or at least a zone delegated to Route53. CoreDNS would give the option to run a CoreDNS instance locally, on the control plane cluster, and interact with it like we would do an external provider.  This could then be used locally, assuming you can add an additional nameserver on your host, to demo features. 

External DNS have a CoreDNS provider which just interacts with an etcd instance, and CoreDNS is configured with zone backed by etcd.

It might be possible to add some kind of ManagedZone support , updating CoreDNS config, but doesn't seem worth while and a single zone for development purposes would work fine.

Reply:
Yeah for local setup or upstream community use not requiring a cloud provider would be a pretty compelling option to get developers to try it out